states.go

```diff
@@ -37,6 +37,17 @@ func ContainsState(s State, other ...State) bool {
 	return false
 }
 
+// StateIndex returns the index of the given state in given states.
+// Returns -1 if state is not found.
+func StateIndex(s State, other ...State) int {
```

Comment:
better to make this function non-exportable and make the `other` argument non-variadic ([]State) .